03-praticas/utils/utils.go

```diff
@@ -0,0 +1,135 @@
+package utils
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+func StringToSliceInt(numString string, hasVirgula bool) ([]int, error) {
+	numArray := []int{}
+	var numStrs []string
+
+	if hasVirgula {
+		numStrs = strings.Split(strings.TrimSpace(numString), ",")
+	} else {
+		numStrs = strings.Split(strings.TrimSpace(numString), "")
+	}
+
+	for _, numStr := range numStrs {
+		num, err := strconv.Atoi(strings.TrimSpace(numStr))
+		if err != nil {
+			return []int{}, errors.New("erro ao converter " + numStr + " para número: " + err.Error() + "\n")
+		}
+
+		numArray = append(numArray, num)
+	}
+
+	return numArray, nil

```

Comment:
Funciona y está perfecto, si pensamos en un array que tal vez envie 10 numeros y tenemos 2 con errores, el proceso saldria al encontrar el 1 error. Luego de corregido el 1 error, volveria el proceso encontrando el 2 error. Pensando en el usuario algunas veces es mejor informar todo de una.

```suggestion

	var errorMsgs []string

	for _, numStr := range numStrs {
		num, err := strconv.Atoi(strings.TrimSpace(numStr))
		if err != nil {
			errorMsgs = append(errorMsgs, fmt.Sprintf("erro ao converter %s para número: %s", numStr, err.Error()))
			continue
		}
		numArray = append(numArray, num)
	}

	if len(errorMsgs) > 0 {
		return numArray, errors.New(strings.Join(errorMsgs, "; "))
	}

	return numArray, nil

```